### PR TITLE
Extract Aspire.Hosting.Valkey.Tests project

### PR DESCRIPTION
**Pull request by eerhardt**
Contributes to #3185
Contributes to #4294
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/4930)

### CODE REVIEW

File: tests/Aspire.Hosting.Valkey.Tests/ValkeyFunctionalTests.cs [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+            if (volumeName is not null)
+            {
+                DockerUtils.AttemptDeleteDockerVolume(volumeName);
+            }
+
+            if (bindMountPath is not null)
+            {
+                try
+                {
+                    File.Delete(bindMountPath);
```
Comment (radical) [inline tests/Aspire.Hosting.Valkey.Tests/ValkeyFunctionalTests.cs:181]: Should this be doing `Directory.Delete`?

File: tests/Aspire.Hosting.Valkey.Tests/ValkeyFunctionalTests.cs [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        Assert.Equal("value", value);
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    [RequiresDocker]
+    public async Task WithDataShouldPersistStateBetweenUsages(bool useVolume)
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(3));
```
Comment (radical) [inline tests/Aspire.Hosting.Valkey.Tests/ValkeyFunctionalTests.cs:62]: unused

File: tests/Aspire.Hosting.Valkey.Tests/ValkeyFunctionalTests.cs [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+namespace Aspire.Hosting.Valkey.Tests;
+
+public class ValkeyFunctionalTests(ITestOutputHelper testOutputHelper)
+{
+    [Fact]
+    [RequiresDocker]
+    public async Task VerifyValkeyResource()
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(3));
```
Comment (radical) [inline tests/Aspire.Hosting.Valkey.Tests/ValkeyFunctionalTests.cs:22]: unused

**Review by radical — APPROVED** [review]
Other than the feedback, LGTM 👍

File: tests/Aspire.Hosting.Valkey.Tests/ValkeyFunctionalTests.cs (hunk 1 above)
Comment (eerhardt) [reply tests/Aspire.Hosting.Valkey.Tests/ValkeyFunctionalTests.cs]: Good call. Looks like we made this mistake in a few places. Fixed them all.

**Comment by radical** [comment]
Closed/re-opened to get changes from `main` which have the fix for Kakfa test failures.

**Pull request merged**